flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/common/config/PulsarConfigUtils.java

```diff
@@ -217,26 +217,7 @@ public static PulsarClient createClient(
      * except the authentication.
      */
     public static PulsarClient createClient(ClientConfigurationData config) {
-        // PulsarClientBuilder don't support using the given ClientConfigurationData directly.
-        Map<String, Object> configMap = configMap(config);
-
-        // These two auth config would be serialized to a secret information by default.
-        configMap.put("authParamMap", config.getAuthParamMap());
-        configMap.put("authParams", config.getAuthParams());
-
-        ClientBuilder clientBuilder = PulsarClient.builder().loadConf(configMap);
-
-        // Set some non-serializable fields.
-        if (config.getAuthentication() != null) {
-            clientBuilder.authentication(config.getAuthentication());
-        }
-        if (config.getClock() != null) {
-            clientBuilder.clock(config.getClock());
-        }
-        if (config.getServiceUrlProvider() != null) {
-            clientBuilder.serviceUrlProvider(config.getServiceUrlProvider());
-        }
-
+        ClientBuilder clientBuilder = new ClientBuilderImpl(config);
```

Comment:
I'm not a big fan of using internal API. Should we try to get some of your changes into Pulsar upstream?
I could imagine that such a Test client would be beneficial also for other Pulsar users.

Reply:
Is this just a way to get rid of the `configMap` method? If so, wouldn't it make sense to rather remove `ClientConfigurationData` from user-facing and just go with `Configuration` all the way?